app/controllers/user.js

```diff
@@ -39,6 +43,41 @@ exports.create = (req, res) => {
     });
 };
 
+exports.generate = (req, res) => {
+  const randomPassword = crypto.randomBytes(5).toString('hex');
+  const cardKey = uuidv4();
+  const userObject = {
+    username: req.body.email
+      .split('@')[0]
+      .match(/[a-zA-Z]+/g)
+      .join(''),
+    cardKey: cardKey,
+  };
+  const user = new User(userObject);
+  return User.register(user, randomPassword)
+    .then(async (createdUser) => {
```

Comment:
Hmmm. This should probably return the `mailHandler`, and not be async.

Also, the `.then(() => res.status(201).json(createdUser.getCleanUser()));` should probably be moved outside to instead have `then(thing).then(thing2)` or async/await?

Reply:
Did you mean something like this
```js
  return User.register(user, randomPassword)
    .then((createdUser) =>
      mailHandler(createdUser.username, req.body.email, randomPassword)
        .then(() => createdUser.getCleanUser())
        .then((user) => res.status(201).json(user))
    )
    .catch(mongoose.Error.ValidationError, (err) => {
    ......
```
or am I totally of the beat?

Reply:
Ahh, didn't notice the data dependence. This works  :+1: